File: tools-manage/invocationlab-admin/src/main/java/org/javamaster/invocationlab/admin/serializer/ByteArrayToJsonSerializer.java
package org.javamaster.invocationlab.admin.serializer;

import com.fasterxml.jackson.core.JsonGenerator;
import com.fasterxml.jackson.databind.JsonSerializer;
import com.fasterxml.jackson.databind.SerializerProvider;

import java.io.IOException;
import java.nio.charset.StandardCharsets;

/**
 * @author yudong
 */
public class ByteArrayToJsonSerializer extends JsonSerializer<byte[]> {
    @Override
    public void serialize(byte[] value, JsonGenerator gen, SerializerProvider serializers) throws IOException {
        if (value == null) {
            gen.writeNull();
        } else {
            gen.writeString("BINARY," + value.length + " Bytes(" + new String(value, StandardCharsets.UTF_8) + ")");
        }
    }
}

File: tools-manage/invocationlab-admin-21/src/main/kotlin/org/javamaster/invocationlab/admin/serializer/ByteArrayToJsonSerializer.kt
package org.javamaster.invocationlab.admin.serializer

import com.fasterxml.jackson.core.JsonGenerator
import com.fasterxml.jackson.databind.JsonSerializer
import com.fasterxml.jackson.databind.SerializerProvider
import java.nio.charset.StandardCharsets

/**
 * @author yudong
 */
class ByteArrayToJsonSerializer : JsonSerializer<ByteArray?>() {

    override fun serialize(value: ByteArray?, gen: JsonGenerator, serializers: SerializerProvider) {
        if (value == null) {
            gen.writeNull()
        } else {
            gen.writeString("BINARY," + value.size + " Bytes(" + String(value, StandardCharsets.UTF_8) + ")")
        }
    }
}
